### DISCUSSION TITLE
pattern_file

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#60] [Announcements]
# 文件读写

https://labview.qizhen.xyz/pattern_file

**Comment by Awhitesu** [comment]
老师您好，一次写入10个有序整数和一次写入一个数组大小为10的有序整数数组，这两种写入二进制文件的方法，我的理解是，写入的内容相同。因为整数和元素为整数的数组，数据类型一致。在二进制的形式下，内容是一样的。影响二进制内容的只有字节顺序，如果字节顺序一致，内容也应该是一致的。

**Reply by ruanqizhen** [reply]
是有一个区别的。像数组这种类型（还有字符串等），它不但包含了显示给用户看的数据：每个元素的数值，还包含了一个不直接显示给用户看的数据：数组的长度。在写二进制文件的时候，这些不显示出来的数据也会被写入文件。
所以如果只想保存数据的话，就不能直接把数组传递给写文件函数。

**Reply by Awhitesu** [reply]
谢谢老师的回复，明白了

**Comment by XXWJ4Zzzz** [comment]
二进制写入中，直接写入整个数组“0~9”，如果读取总数依旧为10，则会显示“10,0,1..,8”，首位显示的将是数组的元素个数，只有将读取总数设为元素数量+1，并删除首位元素，才能得到“0,1,2..,8,9”的正常结果。

**Reply by Awhitesu** [reply]
感谢回复，学到了

**Comment by FuzzyOz** [comment]
老师您好，我在做示波器控制程序的时候，遇到示波器传回的是PNG文件的16进制或RAW，LabVIEW支持这种将PNG数据流转换为PNG图片显示在前面板的操作方式吗？

**Reply by ruanqizhen** [reply]
不支持。也许 LabVIEW 内部有这样的 VI，但是至少公开的 API 我没有见到过。有几个解决方法：
一是可以把数据流存成文件，再使用 LabVIEW 提供的 read png VI 读取数据；
或者，研究一下 PNG 的数据格式，自己把它转换成 LabVIEW 的图片数据，也就是 RGB 三通道的位图数据（BMP 格式）；
也可以找找看有没有第三方的库，可以把 PNG 数据流转换成 BMP 格式的。

**Reply by FuzzyOz** [reply]
好的，感谢老师解惑
